pkg/sql/table.go

```diff
@@ -268,14 +251,16 @@ type uncommittedDatabase struct {
 // as errors, or retries that result in transaction timestamp changes.
 type TableCollection struct {
 	// The timestamp used to pick tables. The timestamp falls within the
-	// validity window of every table in tables.
+	// validity window of every table in leasedTables.
 	timestamp hlc.Timestamp
+
+	// leaseMgr manages acquiring and releasing per-table leases.
+	leaseMgr *LeaseManager
 	// A collection of table descriptor valid for the timestamp.
 	// They are released once the transaction using them is complete.
 	// If the transaction gets pushed and the timestamp changes,
 	// the tables are released.
-	tables []*sqlbase.TableDescriptor
-
+	leasedTables []*sqlbase.TableDescriptor
```

Comment:
Really? Don't we call `LeaseManager.AcquireByName` in `getTableVersion`?

Reply:
Keeping this as is because I'd still rather have some kind of specifier about what these table descriptors are so that it's clear why we shouldn't just merge this slice with `allDescriptors`. Let me know if you have a better name.